plugin/flags_sources/compilation_db.py

```diff
@@ -127,13 +128,17 @@ def _parse_database(self, database_file):
         parsed_db = {}
         unique_list_of_flags = UniqueList()
         for entry in data:
-            file_path = File.canonical_path(entry['file'],
-                                            database_file.folder())
             argument_list = []
 
-            base_path = database_file.folder()
+            file_path = entry['file']
+
             if 'directory' in entry:
                 base_path = entry['directory']
+            else:
+                base_path = database_file.folder()
+
+            if not os.path.isabs(file_path):
```

Comment:
No need for this check, as `File.canonical_path` checks for this

Reply:
Uhm that is needed, as its a change in behaviour,
When file_path is absolute then its selfstanding, nothing is added before it. And just added into the database.
If its not absolut entry['file'] is added after base_path, to make sure it is really the path that is expected according to the spec. 

Or did i understand things not correctly, ?

Reply:
This function makes sense we just normalize the path when it is absolute: https://github.com/niosus/EasyClangComplete/blob/34c3a9720d31b97f104e74eb945ce0775660486d/plugin/tools.py#L259-L261